internal/rbd/replicationcontrollerserver.go

```diff
@@ -517,6 +517,11 @@ func (rs *ReplicationServer) PromoteVolume(ctx context.Context,
 		}
 	}
 
+	err = checkHealthyPrimary(ctx, rbdVol)
```

Comment:
may be I am missing something here: so the check happens and if it fails we return error , I mean this is not a blocking operation?

Reply:
sorry, i didn't get you. can you please explain a bit more?

Reply:
@Madhu-1 I was mainly asking that, once we return an error do we come back here and try again in next iteration? because I was thinking about a scenario where the `up+stopped` was done with a delay in backend , meanwhile this check was performed and failed.

Reply:
once we return the error as it's an RPC call the VR operator will reissue it till the PromoteVolume succeeds. the whole idea of adding this check is to make sure the primary image is healthy before marking it as primary in  VR.